### ISSUE TITLE
Request: add option "export..." which will allow to export to SVG/PNG files

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [fontforge/fontforge#4826]
As the title says. 
Currently we have to use this workaround:
https://stackoverflow.com/a/50127893/878126

**Comment by iorsh** [comment]
Using the scripting language (native or Python) is not a workaround, it's the proper way to achieve what you want. FontForge has a limited amount of command line options, meant to support basic CLI and integration with operating system.

All the complex font-related stuff should be done with scripts.

**Comment by AndroidDeveloperLB** [comment]
@iorsh Sorry but I don't understand what you mean. I only mentioned the link as this is what I've found, not because I succeeded doing it. 
I requested to add a UI-based feature to a UI-based app:
https://fontforge.org/en-US/

Meaning, suppose you find some specific glyphs you want to export. You select them in the window, and then you choose to which format to export: SVG/PNG/TTF.

The command I've written about is a workaround as it's not inside the app itself. Not comfortable to use. 
Please offer a better option for people who can't handle such things. 

I'm sorry if I made you upset for this.

**Comment by iorsh** [comment]
No, you didn't made me upset. Perhaps I expressed myself in an overly harsh manner, sorry about that.

To export a single glyph to SVG/PNG you need to open it in a Character View and go to _File --> Export..._ There is an option to select the export format.

<img width="760" height="689" alt="Image" src="https://github.com/user-attachments/assets/3d777dbf-2b8e-4dcf-81e2-a64ff94f49dd" />

Generation of TTF file with selected glyphs only is more complicated. The easiest way would be to delete all the unselected glyphs (you can achieve this by doing _Element->Select->Invert Selection_, and then _Element->Clear_). After you generated the TTF, just close the font without saving.

**Comment by AndroidDeveloperLB** [comment]
1. Can you please make it work without special operations, meaning just that I have multi-selection of glyphs, and then I choose "export selected to..."? I want for example to export only the flags from [this font file](https://github.com/mozilla/twemoji-colr/releases), either to a new TTF file or to SVG/PNG files. 

2. Would the export work even if FontForge fails to view them? I've noticed various glyphs that aren't showing well...

3. Does it also work in the opposite direction? For example suppose I've found a TTF font file (like [here](https://github.com/mozilla/twemoji-colr/releases)) that I've noticed that it's [missing digits glyphs](https://github.com/mozilla/twemoji-colr/issues/56), can I add them from another font (example [this one](https://fonts.google.com/noto/specimen/Noto+Color+Emoji))?

**Comment by skef** [comment]
@AndroidDeveloperLB this is the issue queue used to report bugs and features in FontForge, where a bug is some way in which the program's behavior doesn't match the documentation, a crash, file corruption, etc. And new feature suggestions are best requested by users who are already familiar with the program. There are more than 1000 open issues, and in any case you are in the best position to optimize your own workflow. If you want to add to what the UI can do you can write an extension that puts an entry in the Tools menu. 

For help using the program you can look at the documentation or try the [FontForge user's mailing list](https://sourceforge.net/p/fontforge/mailman/fontforge-users/).

**Comment by AndroidDeveloperLB** [comment]
@skef Are you saying all that I wrote are fine and possible already?
And that I can ignore the viewing issue if it's about export?

**Comment by skef** [comment]
@AndroidDeveloperLB It's possible to write python code to place selected glyphs into a TTF, and then it's possible to run that code from a menu entry under Tools, and even pop up a dialog asking for the filename to save it in, etc. 

However, would it be a *well-functioning* TTF? Probably not, as there is the entire question of what arguments the TTF will be built with, what substitutions or positioning rules should apply, etc. That sort of thing would normally be handled by a font subsetter, which is something other tools are designed to do and FontForge really isn't (and doesn't need to be).

**Comment by AndroidDeveloperLB** [comment]
@skef Why do you say FontForge isn't designed for such operations?

It says this on the main page of it here:

> FontForge is a free (libre) font editor for Windows, Mac OS X and GNU+Linux. Use it to create, edit and convert fonts in OpenType, TrueType, UFO, CID-keyed, Multiple Master, and many other formats.

https://github.com/fontforge/fontforge?tab=readme-ov-file#fontforge---

As an editor, it should be able to view the content, and as an editor, it should be able to modify existing files and create new files. 

I was told above that it is possible at least some of what I was asking about, just that it is tricky in some cases as it requires more steps.

**Comment by AndroidDeveloperLB** [comment]
@iorsh I tried to export again, but it fails, showing the font file has errors:

<img width="2878" height="1374" alt="Image" src="https://github.com/user-attachments/assets/e9938941-0b0f-47e3-a2f1-f4767cab796b" />

What I tried is as such:

1. Open the TTF file from [here](https://github.com/mozilla/twemoji-colr/releases)

2. Select all digits from 0..9 , and choose to clear them (because this font has issues with digits, as written [here](https://github.com/mozilla/twemoji-colr/issues/56)).

3. File->generate fonts , and choosing where to save.
How can I solve this?
FontForge sadly doesn't seem to view a lot of the glyphs there, so I have no idea what's going on...

**Comment by iorsh** [comment]
The errors you are seeing are minor, and should not affect the font generation. The big issue here is that you are trying to open a color font, and color fonts are not supported - see #677. You wouldn't be able to do anything meaningful with this font using FontForge only.

There are some methods to deal with color fonts using external tools, but they are all quite advanced.
